lib/grape/validations/params_scope.rb

```diff
@@ -94,7 +95,11 @@ def should_validate?(parameters)
       def meets_dependency?(params, request_params)
         return true unless @dependent_on
         return false if @parent.present? && !@parent.meets_dependency?(@parent.params(request_params), request_params)
-        return params.any? { |param| meets_dependency?(param, request_params) } if params.is_a?(Array)
+
+        if params.is_a?(Array)
+          @params_meeting_dependency = params.filter { |param| meets_dependency?(param, request_params) }
```

Comment:
This is memoized, shouldn't it be

```suggestion
          @params_meeting_dependency ||= params.filter { |param| meets_dependency?(param, request_params) }
```

Reply:
I think so. Let’s get rid of the unneeded memorization?

Reply:
but I think it is already not memorized, you mean this line right?
```ruby
@params_meeting_dependency = params.filter { |param| meets_dependency?(param, request_params) }
```
I am already using `=` here, so which memorization you mean?

Reply:
Nevermind, this is fine. I see we need  `@params_meeting_dependency` later. I dislike that we have a `@` in here but it can be called multiple times.